src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -56,22 +75,38 @@ case class GenomeReference(name: String, contigs: Array[Contig], xContigs: Set[S
       s"The contig name for PAR interval `$i' was not found in xContigs `$xContigs' or in yContigs `$yContigs'.")
   }
 
-  val contigIndex: Map[String, Int] = contigs.map(_.name).zipWithIndex.toMap
-  val contigNames: Set[String] = contigs.map(_.name).toSet
+  val contigsIndex: Map[String, Int] = contigs.zipWithIndex.toMap
+  val contigsSet: Set[String] = contigs.toSet
+  val lengthsByIndex: Array[Int] = contigs.map(lengths)
+
+  lengths.foreach { case (n, l) =>
+    if (l <= 0)
+      fatal(s"Contig length must be greater than 0. Contig `$n' has length equal to $l.")
+  }
 
-  val xNotInRef = xContigs.diff(contigNames)
-  val yNotInRef = yContigs.diff(contigNames)
-  val mtNotInRef = mtContigs.diff(contigNames)
+  val xNotInRef = xContigs.diff(contigsSet)
+  val yNotInRef = yContigs.diff(contigsSet)
+  val mtNotInRef = mtContigs.diff(contigsSet)
 
   require(xNotInRef.isEmpty, s"The following X contig names were not found in the reference: `${ xNotInRef.mkString(", ") }'.")
```

Comment:
These errors can be triggered by users, right? So I think they should be "fatal" rather than "required".